capstone/client/src/components/ListPage.js

```diff
@@ -52,6 +54,7 @@ class ListPage extends Component {
     this.getItemTypes();
     
     this.selectedItems = new Set();
```

Comment:
I think one way of doing it might be (but I didn't think about this too much):
update the `this.state.items` to also track if the item has been clicked.
Every time someone clicks on a list button, you will update this list of selected items. In the `render()` function, the checked state would be set by `this.state.items.isChecked` or something like that. So everytime the check changes the UI component would be updated.